config/config.go

```diff
@@ -383,7 +382,10 @@ func getMetadata(suffix string) ([]byte, string, error) {
 func WatchConfig(ctx context.Context) error {
```

Comment:
since this watchconfig is becoming such a critical piecd, i think it is worth add mock tickers and test this method thoroughly in unit test. wdyt?